src/App.tsx

```diff
@@ -1,41 +1,63 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getAllUsers } from './api/api';
+import { User } from './react-app-env';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+const App: React.FC = () => {
+  const [allUsers, setAllUsers] = useState<User[]>([]);
+  const [currentUser, setCurrentUser] = useState('0');
```

Comment:
1. If I'm not mistaken, the `currentUser` contains the user id. So, my recommendation is to name it according to its content 
2. And I think, it's better to add a type here also. 
3. According to the User id type, it should be a number, not a string. So, the default value can't be a string

```suggestion
  const [currentUserId, setCurrentUserId] = useState<User['id']>(0);
```